dashboard/models.py

```diff
@@ -212,24 +211,29 @@ class Course(models.Model):
     def __str__(self):
         return self.name
 
-    @property
-    def course_date_range(self):
+    def determine_date_start(self) -> datetime:
         if self.date_start is not None:
             start = self.date_start
         elif self.term is not None and self.term.date_start is not None:
             start = self.term.date_start
         else:
             logger.info(f"No date_start value was found for course {self.name} ({self.canvas_id}) or term; setting to current date and time")
             start = datetime.now(pytz.UTC)
+        return start
+
+    def determine_date_end(self, start: Union[datetime, None] = None) -> datetime:
         if self.date_end is not None:
             end = self.date_end
         elif self.term is not None and self.term.date_end is not None:
             end = self.term.get_correct_date_end()
         else:
-            logger.info(f"No date_end value was found for course {self.name} ({self.canvas_id}) or term; setting to two weeks past start date.")
-            end = start + timedelta(weeks=2)
-        DateRange = namedtuple("DateRange", ["start", "end"])
-        return DateRange(start, end)
+            logger.info(
+                f"No date_end value was found for course {self.name} ({self.canvas_id}) or term; " +
+                "setting to two weeks past start date."
+            )
+            date_start = start if start else self.determine_date_start()
+            end = date_start + timedelta(weeks=2)
```

Comment:
I felt the assignment of the end date could be done with `date_end` variable if you like as it is done for `date_start` just for consistency.

Reply:
Okay, I made it a bit more consistent. I was just trying to use a different name than the `start` parameter. https://github.com/tl-its-umich-edu/my-learning-analytics/pull/1431/commits/1dc178a66d426a53d6d71c7abb7b4037df5d8fad

Reply:
Nice!